_source/_posts/2023-10-12-okta-documentation-chatbot.md

```diff
@@ -0,0 +1,155 @@
+---
+layout: blog_post
+title: "How I Built an Okta Documentation Chatbot in Python"
+author: tanish-kumar
+by: internal-contributor
+communities: [python]
+description: "How I built a Python chatbot to quickly access information from Okta's Developer Documentation."
+tags: [python-chatbot, python, okta-documentation, ai, automation]
+tweets:
+- ""
+image: blog/okta-documentation-chatbot/oktanaut.jpeg
+type: awareness
+github: https://github.com/oktadev/okta-python-chatbot-example
+---
+
+In today's fast-paced world of technology, developers must navigate through detailed API documentation to integrate with software solutions. As a Developer Support Intern at Okta, I noticed developers underutilizing [Okta's Developer Documentation] (developer.okta.com). To help them benefit from the wealth of information in these docs, I was inspired to build a tool using OpenAI. Leveraging ChatGPT, I built ✨Oktanaut✨, a versatile Python chatbot running on Jupyter Notebook that makes access to information on Okta's Developer Documentation simple and straightforward.
+
+{% img blog/okta-documentation-chatbot/oktadevforumsdk.jpeg alt:"OktaDev Forum and SDK logo" height:"400" %}{: .center-image } 
+– Image source: <cite>[Okta Developer GitHub](https://raw.githubusercontent.com/oktadev/.github/main/images/okta-dev-header.png)</cite> 
+
+## Evolution of Oktanaut, a Python Chatbot 
+I developed two versions of Oktanaut. The first version was a specific approach, offering greater precision rather than general knowledge. It built this chatbot using LlamaIndex and trained it on the Okta Developer Documentation. While this version generated more accurate responses due to its focused training, it could not handle diverse questions.
```

Comment:
```suggestion
I developed two versions of Oktanaut. The first version was a specific approach, offering greater precision rather than general knowledge. I built the prototype chatbot using LlamaIndex and trained it on the Okta Developer Documentation. While this version generated accurate responses due to its focused training, it could not handle diverse questions.
```